content/starter-configs/examples/ruby-on-rails.md

```diff
@@ -0,0 +1,108 @@
+---
+title: "Ruby on Rails"
+date: 2020-07-31T17:00:00-04:00
+weight: 6
+---
+
+This Ruby on Rails config example starts with a Tugboat Debian image, installs Ruby, and uses a Tugboat PostgreSQL
+image, adding and then removing permissions to create a database Rails-style. You'll need to do some customizing for
+your app, but this should get you started.
+
+## Configure Tugboat
+
+The Tugboat configuration is managed by a [YAML file](/setting-up-tugboat/create-a-tugboat-config-file/) at
+`.tugboat/config.yml` in the git repository. Here's a basic Ruby on Rails configuration you can use as a starting point,
+with comments to explain what's going on:
+
+```yaml
+services:
+  # What to call the service hosting the site.
+  rails:
+    # Use a Tugboat Debian image as a starting point because it contains tools
+    # and configurations to help them work well with Tugboat
+    image: tugboatqa/debian
+
+    # Set this as the default service. This does a few things
+    #   1. Clones the git repository into the service container
+    #   2. Exposes port 80 to the Tugboat HTTP proxy
+    #   3. Routes requests to the Preview URL to this service
+    default: true
+
+    # Don't build the Ruby service until the database service is started
+    depends: database
+
+    # A set of commands to run while building this service
+    commands:
+      # Commands that set up the basic preview infrastructure
+      init:
+        # Check for updates to apt-get, and then use it to install Ruby
+        - apt-get update
+        - apt-get install ruby ruby-dev
+
+        # Install a minimal set of PostgreSQL binaries and headers to interact with the database
+        - apt-get install libpq-dev
+
+        # Install the gems required for the project
+        - gem update --system
+        - gem install bundler
+        - bundle install
+
+        # Clean up apt artifacts to keep the Preview small
+        - apt-get clean
+        - rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+      # Commands that import files, databases,  or other assets. When an
+      # existing preview is refreshed, the build workflow starts here,
+      # skipping the init step, because the results of that step will
+      # already be present.
+      update:
+        # A common practice in many Rails projects is to run this command to
+        # set up a database. This step is why the 'database' service has an 'init'
+        # command to give the default Tugboat user permission to create the database.
+        # If you're not creating a database here, you don't need the 'init' and
+        # 'online' commands in the `database` service. If you're connecting to a remote
+        # database, you won't need this.
+        - bin/rails db:setup
```

Comment:
Good question! Just checked it - I was able to Refresh without it blowing up - and then Kay confirmed for me how this command operates.

`db:setup` drops the database, and then recreates it. Had a chat with him about whether this is where Rails devs would most likely want to work with the db, or whether it would be in `init`. He suggests if using a seed db, that data tends to be pretty static and doing it in `init` once would be fine. But he also says not a lot of devs have the discipline to use seed databases to test against, and therefore are likely to be using production data, and in that case, they might want this behavior (i.e. creating a new, updated db during refresh). 

What do you think @bchavet - should I add some verbiage in the comment about doing this in `init` if you have a static seed db, or leaving it here if you want it to create a new db every time there's a Refresh?

Reply:
The `update` section is where we recommend pulling in fresh data for the database, so it makes sense to leave it here regardless of whether the user uses a seed database or not.

I just was unsure of how it worked.